### PR TITLE
Update templates to support license autoload.

### PR DESCRIPTION
**Pull request by thisisnotashwin**
Changes proposed in this PR:
- Make updates required to support license auto loading.
- Ensure backwards compatibility for consul versions where license job is required to run.

TODO:
- Update acceptance tests to use autoLoad for consul 1.10+ (separate PR) 

How I've tested this PR:
- Run the acceptance tests with a Consul License set as a secret and autoloading enabled with ACLs enabled and disabled.

How I expect reviewers to test this PR:
- Code review
- Status of CI pipeline

Checklist:
- [x] Bats tests added
- [ ] CHANGELOG entry added (*HashiCorp engineers only, community PRs should not add a changelog entry*)

### CODE REVIEW

File: templates/enterprise-license-podsecuritypolicy.yaml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {{- if (or (and (ne (.Values.server.enabled | toString) "-") .Values.server.enabled) (and (eq (.Values.server.enabled | toString) "-") .Values.global.enabled)) }}
-{{- if (and .Values.server.enterpriseLicense.secretName .Values.server.enterpriseLicense.secretKey) }}
+  {{- if (and (and .Values.server.enterpriseLicense.secretName .Values.server.enterpriseLicense.secretKey) (not .Values.server.enterpriseLicense.enableLicenseAutoload)) }}
```
Comment (lkysow) [inline templates/enterprise-license-podsecuritypolicy.yaml:2]:
```suggestion
{{- if (and (and .Values.server.enterpriseLicense.secretName .Values.server.enterpriseLicense.secretKey) (not .Values.server.enterpriseLicense.enableLicenseAutoload)) }}
```

File: templates/enterprise-license-role.yaml [hunk 2]
```diff
@@ -1,5 +1,5 @@
 {{- if (or (and (ne (.Values.server.enabled | toString) "-") .Values.server.enabled) (and (eq (.Values.server.enabled | toString) "-") .Values.global.enabled)) }}
-{{- if (and .Values.server.enterpriseLicense.secretName .Values.server.enterpriseLicense.secretKey) }}
+  {{- if (and (and .Values.server.enterpriseLicense.secretName .Values.server.enterpriseLicense.secretKey) (not .Values.server.enterpriseLicense.enableLicenseAutoload)) }}
```
Comment (lkysow) [inline templates/enterprise-license-role.yaml:2]:
```suggestion
{{- if (and (and .Values.server.enterpriseLicense.secretName .Values.server.enterpriseLicense.secretKey) (not .Values.server.enterpriseLicense.enableLicenseAutoload)) }}
```

**Review by lkysow — COMMENTED** [review]
Looking good. Some template change recommendations.

File: templates/client-daemonset.yaml [hunk 3]
```diff
@@ -127,6 +127,12 @@ spec:
         {{- if .Values.global.acls.manageSystemACLs }}
         - name: aclconfig
           emptyDir: {}
+        {{- else }}
+        {{- if (and (and .Values.server.enterpriseLicense.secretName .Values.server.enterpriseLicense.secretKey) .Values.server.enterpriseLicense.enableLicenseAutoload) }}
```
[the remark this answers is not in the document]
Comment (thisisnotashwin) [reply templates/client-daemonset.yaml]: Nice catch!! TIL

**Review by lkysow — APPROVED** [review]
yaml looks good. accept tests are failing but didn't look at why

**Review by ndhanushkodi — APPROVED** [review]
Looks great, the tests even helped me understand a change I was confused about for a bit!! Nice work!!

**Comment by thisisnotashwin** [comment]
> yaml looks good. accept tests are failing but didn't look at why

had a look into it and it turns out we need the license to be applied in the primary and secondary datacenters in the mesh gateway tests. im a little surprised that wasnt previously the case but pushed a fix for it.

**Pull request merged**